overlord/snapstate/snapmgr_test.go

```diff
@@ -551,7 +551,10 @@ func (s *snapmgrTestSuite) TestUpdateUndoIntegration(c *C) {
 			op:   "link-snap.failed",
 			name: "/snap/some-snap/11",
 		},
-		// no unlink-snap here is expected!
```

Comment:
@zyga the undo mechanism takes care only of done tasks, so a failing task should consider cleaning up/undoing its job on errors on its own if it make sense, I added such code to doLinkSnap now (the code is a bit ugly actually, waiting for more cases to find a pattern there)